File: backends/aspnet-core/Services/VehicleService.cs
using LogisticsAspNetCore.Models;
using LogisticsAspNetCore.Repositories;

namespace LogisticsAspNetCore.Services;

public class VehicleService(IVehicleRepository vehicleRepository) : IVehicleService
{
    public IReadOnlyList<Vehicle> GetAllVehicles()
    {
        return vehicleRepository.FindAll();
    }

    public Vehicle? GetVehicleById(long id)
    {
        return vehicleRepository.FindById(id);
    }

    public Vehicle AddVehicle(Vehicle vehicle)
    {
        return vehicleRepository.Create(vehicle);
    }

    public Vehicle? UpdateVehicle(long id, Vehicle vehicle)
    {
        return vehicleRepository.Update(id, vehicle);
    }

    public bool RemoveVehicleById(long id)
    {
        return vehicleRepository.DeleteById(id);
    }
}

File: backends/laravel/app/Services/VehicleService.php
<?php

namespace App\Services;

use App\Models\Vehicle;
use App\Repositories\VehicleRepository;
use Illuminate\Database\Eloquent\Collection;

class VehicleService
{
    public function __construct(private readonly VehicleRepository $vehicleRepository)
    {
    }

    public function getAllVehicles(): Collection
    {
        return $this->vehicleRepository->findAll();
    }

    public function getVehicleById(int $id): ?Vehicle
    {
        return $this->vehicleRepository->findById($id);
    }

    public function addVehicle(array $data): Vehicle
    {
        return $this->vehicleRepository->create($data);
    }

    public function updateVehicle(Vehicle $vehicle, array $data): Vehicle
    {
        return $this->vehicleRepository->update($vehicle, $data);
    }

    public function removeVehicle(Vehicle $vehicle): void
    {
        $this->vehicleRepository->delete($vehicle);
    }
}
